app/models/invitation.rb

```diff
@@ -60,12 +60,18 @@ def sent?
   def notify_invitee
     # TODO: maybe subclass to Invitation and ProviderInvitation
     if account.provider?
-      ProviderInvitationMailer.invitation(self).deliver_now
+      ProviderInvitationMailer.register_observer(InvitationDeliveryObserver)
+      ProviderInvitationMailer.invitation(self).deliver_later
     else
-      InvitationMailer.invitation(self).deliver_now
+      InvitationMailer.register_observer(InvitationDeliveryObserver)
+      InvitationMailer.invitation(self).deliver_later
     end
+  end
 
-    update_column(:sent_at, Time.zone.now)
+  class InvitationDeliveryObserver
+    def self.delivered_email()
+      update_column(:sent_at, Time.zone.now)
```

Comment:
Does this suggestion work? If so, this is better.
```suggestion
      update!(sent_at: Time.zone.now)
```